mypy/plugins/attrs.py

```diff
@@ -656,6 +658,20 @@ def _add_init(ctx: 'mypy.plugin.ClassDefContext', attributes: List[Attribute],
     adder.add_method('__init__', args, NoneType())
 
 
+def _add_attrs_magic_attribute(ctx: 'mypy.plugin.ClassDefContext') -> None:
+    attr_name = '__attrs_attrs__'
+    attr_type = ctx.api.named_type('__builtins__.tuple', [
+        ctx.api.named_type('attr.Attribute'),
```

Comment:
@ilevkivskyi `ctx.api.named_type` is for builtin types, but what i should use instead of it?

Reply:
It is not for builtin types, it is for types that are guaranteed to be present (i.e. loaded).

Reply:
@ilevkivskyi, and how to ensure that `attr.Attribute` is loaded? Or maybe need to use `Any` if it is not loaded?

Reply:
They should be loaded, because the `attr` module must be analyzed if this plugin is called. The tests fail because they run with partial stubs, not the real ones, see for example https://github.com/python/mypy/blob/master/test-data/unit/lib-stub/attr.pyi, `Attribute` is not present there.

Also the latter is generic, so I would explicitly add a type argument here (even if it is `Any`).

Reply:
> They should be loaded, because the attr module must be analyzed if this plugin is called.

I have a problem with different styles of import of `attr`.

This example works fine:
```python
import attr

@attr.s
class Foo:
    pass

class AttrsProtocol(Protocol):
    __attrs_attrs__: Tuple[attr.Attribute, ...]

def foo(x: AttrsProtocol) -> None:
    pass

foo(Foo())
```
But if I change import to:
```python
from attr import attrs, Attribute

@attrs
class Foo:
    pass
```
I get exception:
```
  File "/Users/tkukushkin/Projects/mypy/mypy/plugins/attrs.py", line 664, in _add_attrs_magic_attribute
    ctx.api.named_type('attr.Attribute'),
  File "/Users/tkukushkin/Projects/mypy/mypy/semanal.py", line 4143, in named_type
    assert sym, "Internal error: attempted to construct unknown type"
AssertionError: Internal error: attempted to construct unknown type
```
Some tests fail because of this.

> The tests fail because they run with partial stubs, not the real ones, see for example https://github.com/python/mypy/blob/master/test-data/unit/lib-stub/attr.pyi, `Attribute` is not present there.

Yes, I already added `Attribute` to stub, but not pushed yet.

Reply:
Well, I just looked at the code of `named_type()` and it is horrible, it uses `lookup_qualified()` (which is just a normal symbol lookup following Python name resolution rules) so I wonder how it ever worked. You can try using `named_type_or_none()` instead, that one looks reasonable.

Another option is to actually fix `named_type()`, but it may be not the best task as a first contribution.